### PR TITLE
PATCH /events/:id

### LINKED ISSUE #25: Implement `PATCH /events/:id`
## Tasks:

- Implement `PATCH /events/:id`
- Add documentation and unit testing.

### PR DESCRIPTION
**Pull request by BerkAkipek**
closes #25.

### CODE REVIEW

File: src/controllers/__tests__/events.test.js [hunk 1]
```diff
@@ -334,6 +334,25 @@ describe('Testing events for routes require auth controls', () => {
     });
   });
 
+  it('PATCH /events/:id should update the event', async () => {
+    const ngo = await createNgo();
+    const ngoCookie = createToken(ngo);
+    events[0].ngo = ngo._id;
```
Comment (yamanrajab90) [inline src/controllers/__tests__/events.test.js:340]: Great work! 
test has passed, and good job keeping consistency with other tests.

File: src/controllers/events.js [hunk 2]
```diff
@@ -71,6 +71,16 @@ exports.applyToEvent = async (req, res, next) => {
   return res.status(200).json({ success: true, data: event });
 };
 
+
+exports.updateEvent = async (req, res, next) => {
+  const event = await Event.findById(req.params.id);
```
Comment (yamanrajab90) [inline src/controllers/events.js:76]: The controller has been tested through postman, it works fine.

File: src/middlewares/validatorSchemas.js [hunk 3]
```diff
@@ -145,9 +145,43 @@ const validateAddEvent = [
   body('topic').not().isEmpty().withMessage('Topic should not be empty'),
 ];
 
+const validateUpdateEvent = [
+  body('avatar')
+    .not()
+    .isEmpty()
+    .withMessage('Avatar should not be empty')
+    .optional(),
```
Comment (yamanrajab90) [inline src/middlewares/validatorSchemas.js:153]: good job following Muslim's requested changes

File: src/routes/events.js [hunk 4]
```diff
@@ -1,7 +1,10 @@
 const router = require('express').Router();
 const { validate } = require('../middlewares/bodyValidator');
 const eventsControllers = require('../controllers/events');
-const { validateAddEvent } = require('../middlewares/validatorSchemas');
+const {
+  validateAddEvent,
```
Comment (yamanrajab90) [inline src/routes/events.js:5]: good job following Muslim's requested changes

File: swagger.json [hunk 5]
```diff
@@ -871,6 +871,64 @@
           }
         }
       }
+    },
+    "events/{id}": {
+      "patch": {
```
Comment (yamanrajab90) [inline swagger.json:876]: APIDocumentation inserted at the right place.

**Review by yamanrajab90 — APPROVED** [review]

**Comment by yamanrajab90** [comment]
@BerkAkipek can you please inform @abdulhafiz96 to delete the topic from `validateUpdateEvent`  after your pull request is closed. 
or if someone can suggest a different strategy to reduce the work load.(e.g. @yamanrajab90 can replace  the topic when implementing tags.

File: src/controllers/events.js (hunk 2 above)
Comment (muslimomar) [inline src/controllers/events.js:76]: What if the event is null? You should check for that

**Review by muslimomar — CHANGES_REQUESTED** [review]
Good job!

**Pull request closed without merging**